log_outgoing_requests/utils.py

```diff
@@ -0,0 +1,155 @@
+import logging
+from typing import TYPE_CHECKING, Iterable, Optional, Union
+
+from django.conf import settings
+
+from requests import Request, Response
+
+if TYPE_CHECKING:  # pragma: nocover
+    from .models import OutgoingRequestsLogConfig
+
+
+logger = logging.getLogger(__name__)
+
+
+#
+# Admin utilities
+#
+def decode(content: memoryview, encoding: str) -> Union[str, bytes]:
+    """
+    Convert `memoryview` for display.
+
+    :returns: `str` if `encoding` is provided, `bytes` otherwise.
+    """
+
+    content = bytes(content)
+    if encoding:
+        try:
+            content = content.decode(encoding)
+        except UnicodeDecodeError as ex:
+            logger.warning("Some binary data could not be decoded (%s)" % ex)
```

Comment:
good to log this, I'd also fall back to decode it with `errors="replace"` as mentioned earlier, so then you should be able to have a single return type of `str`